src/extension/options-page/DashboardDialogs/Persona.tsx

```diff
@@ -250,6 +252,21 @@ export function PersonaImportDialog() {
         }
     }
 
+    const importFromQR = (str: string) => {
+        Promise.resolve()
+            .then(() => UpgradeBackupJSONFile(decompressBackupFile(str)))
+            .then(object => Services.Welcome.restoreBackup(object!))
+            .then(() => setRestoreState('success'))
+            .catch(() => setRestoreState('failed'))
+    }
+
+    function QR() {
+        const shouldRenderQRComponent = state[0] === 2 && !restoreState
+        return shouldRenderQRComponent ? (
+            <QRScanner onError={() => setRestoreState('failed')} scanning width="100%" onResult={importFromQR} />
```

Comment:
Ah, does it support iOS? For iOS, `<WKWebkitQR onScan={restore} onQuit={() => setTab(0)} />`